### PR TITLE
[controller] Fix CommandSender initializer

### LINKED ISSUE #5788: Enabling interaction model causes crash with null pointer
#### Problem
STR:
1 - set chip_enable_interaction_model = true in src/lib/core/core.gni

2 - build examples/chip-tool

3 - ensure you have a paired device

4 - run with arguments onoff on 1

Result: crash with null pointer. The line in OnOffCluster::On in CHIPClusters.cpp:

app::Command * ZCLcommand             = mDevice->GetCommandSender();

returns null then the null pointer gets used.

 #### Proposed Solution
call mDevice->InitCommandSender() at the appropriate place to ensure the sender is not null.

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem

STR:
1 - set chip_enable_interaction_model = true in src/lib/core/core.gni

2 - build examples/chip-tool

3 - ensure you have a paired device

4 - run with arguments onoff on 1

Result: crash with null pointer. The line in OnOffCluster::On in CHIPClusters.cpp:

app::Command * ZCLcommand = mDevice->GetCommandSender();

returns null then the null pointer gets used.

 #### Summary of Changes

Initialize CommandSender with Device::Init, instead OnRendezvousComplete event.

 Fixes #5788

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

File: src/controller/CHIPDevice.cpp [hunk 1]
```diff
@@ -371,5 +371,17 @@ void Device::AddReportHandler(EndpointId endpoint, ClusterId cluster, AttributeI
...
 
+void Device::InitCommandSender()
+{
+    if (mCommandSender != nullptr)
+    {
+        mCommandSender->Shutdown();
+        mCommandSender = nullptr;
+    }
+#if CHIP_ENABLE_INTERACTION_MODEL
+    chip::app::InteractionModelEngine::GetInstance()->NewCommandSender(&mCommandSender);
```
Comment (yufengwangca) [inline src/controller/CHIPDevice.cpp:382]: We should log the error if NewCommandSender return failure

File: src/controller/CHIPDevice.h [hunk 2]
```diff
@@ -372,6 +361,13 @@ class DLL_EXPORT Device
      */
     CHIP_ERROR LoadSecureSessionParametersIfNeeded(bool & didLoad);
 
+    /**
+     * @brief
+     *   Initialize internal command sender, required for sending commands over interaction model.
```
Comment (yufengwangca) [inline src/controller/CHIPDevice.h:366]: Remove the extra space before 'Initialize'

File: src/controller/CHIPDevice.cpp (hunk 1 above)
Comment (jelderton) [reply src/controller/CHIPDevice.cpp:382]: +1

**Review by jmartinez-silabs — APPROVED** [review]

File: src/controller/CHIPDevice.cpp (hunk 1 above)
Comment (erjiaqing) [reply src/controller/CHIPDevice.cpp]: Done, thanks

File: src/controller/CHIPDevice.h (hunk 2 above)
Comment (erjiaqing) [reply src/controller/CHIPDevice.h:366]: Done

**Pull request merged**